File: infrastructure/src/main/java/com/study/platform/domain/user/infrastructure/UserJpaRepository.java
package com.study.platform.domain.user.infrastructure;

import com.study.platform.domain.user.model.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Collection;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface UserJpaRepository extends JpaRepository<User, UUID> {

    Optional<User> findByKakaoId(String kakaoId);

    Optional<User> findByNickname(String nickname);

    List<User> findAllByNicknameIn(Collection<String> nicknames);
}

File: infrastructure/src/main/kotlin/com/study/platform/domain/user/infrastructure/UserJpaRepository.kt
package com.study.platform.domain.user.infrastructure

import com.study.platform.domain.user.model.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface UserJpaRepository : JpaRepository<User, UUID> {

    fun findByKakaoId(kakaoId: String): User?

    fun findByNickname(nickname: String): User?

    fun findAllByNicknameIn(nicknames: Collection<String>): List<User>
}
